File: scratch/check_blog_images.py
import urllib.request
import ssl
import re

ctx = ssl.create_default_context()
ctx.check_hostname = False
ctx.verify_mode = ssl.CERT_NONE

with open('js/data.js', 'r', encoding='utf-8') as f:
    content = f.read()

posts = re.findall(r'id:\s*(\d+),[\s\S]*?title:\s*"([^"]+)",[\s\S]*?category:\s*"([^"]+)",[\s\S]*?categories:\s*(\[[^\]]+\]),[\s\S]*?image:\s*"([^"]+)"', content)

print(f"Total blog posts found: {len(posts)}\n")
for pid, title, cat, cats, img in posts:
    try:
        req = urllib.request.Request(img, headers={'User-Agent': 'Mozilla/5.0'})
        res = urllib.request.urlopen(req, timeout=5, context=ctx)
        status = res.status
    except Exception as e:
        status = f"ERROR: {e}"
    print(f"Post #{pid}: \"{title}\"")
    print(f"  Category: {cat} | Categories: {cats}")
    print(f"  Image: {img} -> [{status}]\n")

File: scratch/check_blog_images.js
const fs = require('fs');

const dataJs = fs.readFileSync('js/data.js', 'utf8');
global.window = {};
eval(dataJs);

const posts = global.window.AppData.blogPosts;
console.log('Total blog posts:', posts.length);

const imageCounts = {};
posts.forEach(p => {
  console.log(`[Blog ${p.id}] "${p.title}" (${p.category}) -> ${p.image}`);
  imageCounts[p.image] = (imageCounts[p.image] || 0) + 1;
});

console.log('\n--- Duplicate Check in Blog Posts ---');
let hasDupes = false;
Object.keys(imageCounts).forEach(img => {
  if (imageCounts[img] > 1) {
    hasDupes = true;
    const matching = posts.filter(p => p.image === img);
    console.error(`❌ Duplicate Image (${imageCounts[img]} posts): ${img}`);
    matching.forEach(m => console.error(`   - ID ${m.id}: "${m.title}"`));
  }
});

if (!hasDupes) {
  console.log('✅ All blog post images are 100% unique!');
}
